### DISCUSSION TITLE
Find out if speakers are paired ?

### DISCUSSION THREAD

**Discussion by BastelBaus** [avantrec/soco-cli#14] [Q&A] [answered]
I looked now quite a bit in the net but did not find a solution to check if speakers are paired or not. Anyone has a hint for me ?
BR,
BastelBaus

**Comment by pwt** [comment]
The action `sysinfo` provides a table of information about the status of all the speakers in the system. If a speaker has 'State' `grouped`, and 'Visible' is `No`, then it is paired or bonded. The master speaker for the paired/bonded arrangement is the one denoted by 'CoOrd IP'.

So, this lets one infer what grouped/paired/bonded state a speaker has.

(It's possible you're actually interested in whether speakers are **grouped** rather than **paired**? If so, that can be done using the `groups` action, which shows the list of group coordinator speakers, followed by any other speakers that are currently in their group.)

It would also be easy to provide a direct action that reported on whether a specified speaker was paired or bonded (or grouped). I'm not sure of your use case ... if you let me know what you're trying to do, I can think about the best way of supporting it.

**Comment by pwt** [comment, marked as answer]
Version 0.3.17 includes a `groupstatus` action that attempts to report on a speaker's paired/bonded/grouped state. Let me know if this is the kind of thing you're looking for.
